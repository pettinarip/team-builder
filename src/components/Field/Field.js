import React from 'react'
import PropTypes from 'prop-types'
import Layout from '../Layout'

const Field = ({ layout, playersPositions, addPlayer, removePlayer }) => {
  return (
    <div className='field'>
      {layout &&
        <Layout
          config={layout.config}
          playersPositions={playersPositions}
          addPlayer={addPlayer}
          removePlayer={removePlayer}
        />}
    </div>
  )
}

Field.propTypes = {
  layout: PropTypes.object,
  playersPositions: PropTypes.object,
  addPlayer: PropTypes.func.isRequired,
  removePlayer: PropTypes.func.isRequired
}

export default Field
