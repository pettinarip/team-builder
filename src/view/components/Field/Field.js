import React from 'react'
import PropTypes from 'prop-types'
import Layout from '../Layout'

const Field = ({ layout, playersPositions, addPlayer, removePlayer, readOnly }) => {
  return (
    <div className='field'>
      {layout &&
        <Layout
          config={layout.config}
          playersPositions={playersPositions}
          addPlayer={addPlayer}
          removePlayer={removePlayer}
          readOnly={readOnly}
        />}
    </div>
  )
}

Field.propTypes = {
  layout: PropTypes.object,
  playersPositions: PropTypes.object,
  addPlayer: PropTypes.func.isRequired,
  removePlayer: PropTypes.func.isRequired,
  readOnly: PropTypes.bool
}

export default Field
