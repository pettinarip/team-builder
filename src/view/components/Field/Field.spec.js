import React from 'react'
import { shallow } from 'enzyme'
import Field from './Field'

describe('Field component', () => {
  it('should render the component properly', () => {
    const component = mountField({
      layout: {id: 1, config: [4, 4, 2]},
      playersPositions: {1: 1}
    })

    expect(component).toMatchSnapshot()
  })

  it('should not render layout component if no layout is passed', () => {
    const component = mountField({
      playersPositions: {1: 1}
    })

    expect(component).toMatchSnapshot()
  })
})

function mountField (props = {}) {
  const propsToUse = {
    addPlayer () {},
    removePlayer () {},
    layout: undefined,
    playersPositions: {},
    ...props
  }

  return shallow(<Field {...propsToUse} />)
}
