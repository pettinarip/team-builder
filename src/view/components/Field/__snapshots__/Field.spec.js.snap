// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component should not render layout component if no layout is passed 1`] = `
<div
  className="field"
>
  <DragLayer(PositionLayer) />
</div>
`;

exports[`Field component should render the component properly 1`] = `
<div
  className="field"
>
  <DropTarget(Layout)
    config={
      Array [
        Object {
          "x": 10,
          "y": 10,
        },
        Object {
          "x": 10,
          "y": 20,
        },
        Object {
          "x": 10,
          "y": 30,
        },
      ]
    }
    playersPositions={
      Object {
        "1": 1,
      }
    }
    switchPosition={[Function]}
    updatePosition={[Function]}
  />
  <DragLayer(PositionLayer) />
</div>
`;
